framework/include/utils/SerializerGuard.h

```diff
@@ -0,0 +1,27 @@
+#ifndef SERIALIZERGUARD_H
+#define SERIALIZERGUARD_H
+
+namespace libMesh
+{
+namespace Parallel
+{
+class Communicator;
+}
+}
+
+/**
+ * A scope guard that guarantees that whatever happens between when it gets created and when it is
+ * destroyed is done "serially" (each MPI rank will run in turn starting from 0)
+ */
+class SerializerGuard
```

Comment:
this is more elegant than the beginSerial endSerial stuff I added a white ago. Let's remove that.

Reply:
This is what I mean: https://github.com/idaholab/moose/blob/94266fac64014806b1e166f4673a0bd9e908ae68/framework/include/utils/MooseUtils.h#L96

Reply:
This class actually uses the `serialBegin()` and `serialEnd()` functions.  I think it's worth having both (for cases where a scope guard isn't the right answer).